core/src/main/java/org/apache/carbondata/core/geo/GeoHashImpl.java

```diff
@@ -0,0 +1,163 @@
+                      CarbonCommonConstants.INDEX_HANDLER, TARGET_DATA_TYPE));
+    }
+    /* Set the generated column data type as long */
+    properties.put(TARGET_DATA_TYPE, "long");
+    String ORIGIN_LATITUDE = commonKey + "originlatitude";
+    String originLatitude = properties.get(ORIGIN_LATITUDE);
+    if (originLatitude == null) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. Must specify %s property.",
+                      CarbonCommonConstants.INDEX_HANDLER, ORIGIN_LATITUDE));
+    }
+    String MIN_LONGITUDE = commonKey + "minlongitude";
+    String MAX_LONGITUDE = commonKey + "maxlongitude";
+    String MIN_LATITUDE = commonKey + "minlatitude";
+    String MAX_LATITUDE = commonKey + "maxlatitude";
+    String minLongitude = properties.get(MIN_LONGITUDE);
+    String maxLongitude = properties.get(MAX_LONGITUDE);
+    String minLatitude = properties.get(MIN_LATITUDE);
+    String maxLatitude = properties.get(MAX_LATITUDE);
+    if (minLongitude == null || maxLongitude == null
+            || minLatitude == null || maxLatitude == null) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. Must specify %s, %s, %s and %s properties.",
+                      CarbonCommonConstants.INDEX_HANDLER, MIN_LONGITUDE, MAX_LONGITUDE,
+                      MIN_LATITUDE, MAX_LATITUDE));
+    }
+    String GRID_SIZE = commonKey + "gridsize";
+    String gridSize = properties.get(GRID_SIZE);
+    if (gridSize == null) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. %s property must be specified.",
+                      CarbonCommonConstants.INDEX_HANDLER, GRID_SIZE));
+    }
+    String CONVERSION_RATIO = commonKey + "conversionratio";
+    String conversionRatio = properties.get(CONVERSION_RATIO);
+    if (conversionRatio == null) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. %s property must be specified.",
+                      CarbonCommonConstants.INDEX_HANDLER, CONVERSION_RATIO));
+    }
+
+    // TODO: Fill the values to the instance fields
+  }
+
+  /**
+   * Generates the GeoHash ID column value from the given source columns.
+   * @param sources Longitude and Latitude
+   * @return Returns the generated hash id
+   * @throws Exception
+   */
+  @Override
+  public String generate(List<?> sources) throws Exception {
+    if (sources.size() != 2) {
+      throw new RuntimeException("Source columns list must be of size 2.");
+    }
+    if (!(sources.get(0) instanceof Long) || !(sources.get(1) instanceof Long)) {
+      throw new RuntimeException("Source columns must be of Long type.");
+    }
+    //TODO: generate geohashId
+    return String.valueOf(0);
+  }
+
+  /**
+   * Query processor for GeoHash.
+   * @param polygon
```

Comment:
describe the format of the polygon string

Reply:
This is updated in @MarvinLitt PR #3481